### DISCUSSION TITLE
[OP Stack Deployment] Update Foundry Deployment Contract

### DISCUSSION THREAD

**Discussion by avenbreaks** [ethereum-optimism/developers#440] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Getting Error Logs From Forge Deployment : 

Script : `forge script scripts/Deploy.s.sol:Deploy --private-key $GS_ADMIN_PRIVATE_KEY --broadcast --rpc-url $L1_RPC_URL --slow`

### Logs

```
## Setting up 1 EVM.
Error: 
Failed to get EIP-1559 fees
```

### Additional Information

Adding `--legacy` on script deployment 

`forge script scripts/Deploy.s.sol:Deploy --private-key $GS_ADMIN_PRIVATE_KEY --broadcast --rpc-url $L1_RPC_URL --slow --legacy`

**Comment by Chomtana** [comment]
You should use the branch `op-contracts/v1.3.0` and, after deployment, switch to the branch `op-node/v1.7.7-rc.1` to run op-node, op-batcher, and op-proposer.

The develop branch is known to be unstable.

Although `op-contracts/v1.5.0` has been released, it may contain undocumented features that can cause headaches.

**Reply by avenbreaks** [reply]
I added the `--legacy` flag in foundry and it works fine. but extract artifacts I got a failure that is the loss of .json files in the getting-started folder. tried to backup the data but still same thing.
